docs/errors/usage_errors.md

```diff
@@ -522,6 +522,53 @@ except PydanticUserError as exc_info:
     assert exc_info.code == 'removed-kwargs'
 ```
 
+## Circular reference schema {#circular-reference-schema}
+
+While Pydantic can handle recursive types, it does not allow _directly_ circular references in the schema.
+
+For example, these are valid types:
+
+```py test="skip" lint="skip" upgrade="skip"
+type A = A | int
+
+type B = list[C]
+type C = B | None
+```
+
+while these are not:
+
+```py test="skip" lint="skip" upgrade="skip"
+type A = A
+
+type B = C
+type C = B
+```
+
+So the latter will raise an error:
+
+```py
+from typing_extensions import TypeAliasType
+
+from pydantic import PydanticUserError, TypeAdapter
+
+A = TypeAliasType('A', 'A')
+
+B = TypeAliasType('B', 'C')
+C = TypeAliasType('C', 'B')
+
+
+try:
+    TypeAdapter(A)
+except PydanticUserError as exc_info:
+    assert exc_info.code == 'circular-reference-schema'
+
+
+try:
+    TypeAdapter(B)
+except PydanticUserError as exc_info:
+    assert exc_info.code == 'circular-reference-schema'
+```
+
```

Comment:
I think we can remove this part, we already described the kind of valid and invalid type aliases.